aten/src/ATen/core/function_schema_inl.h

```diff
@@ -169,6 +192,86 @@ inline bool FunctionSchema::isBackwardCompatibleWith(
   return true;
 }
 
+inline bool FunctionSchema::isForwardCompatibleWith(
+    const FunctionSchema& old,
+    std::ostream* why_not) const {
+  if (!(name() == old.name() &&
+        overload_name() == old.overload_name()
+        // we are conservative on is_vararg and is_varret,
+        // since they are only used by internal operators
+        && is_vararg() == old.is_vararg() && is_varret() == old.is_varret() &&
+        returns().size() == old.returns().size() &&
+        arguments().size() >= old.arguments().size())) {
```

Comment:
The case of `arguments.size() < old.arguments().size()` can be FC actually. Think about the case where we remove the last argument, which has a default value.

What do you think?

Reply:
That is true. I will update it